cmd/gpud/command/join.go

```diff
@@ -0,0 +1,190 @@
+package command
+
+import (
+	"archive/tar"
+	"bytes"
+	"compress/gzip"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/urfave/cli"
+
+	"github.com/leptonai/gpud/components/state"
+	"github.com/leptonai/gpud/config"
+	"github.com/leptonai/gpud/log"
+	"github.com/leptonai/gpud/pkg/process"
+)
+
+func cmdJoin(cliContext *cli.Context) (retErr error) {
+	rootCtx, rootCancel := context.WithTimeout(context.Background(), 3*time.Minute)
+	defer rootCancel()
+	endpoint := cliContext.String("endpoint")
+	clusterName := cliContext.String("cluster-name")
+	publicIP := cliContext.String("public-ip")
+	provider := cliContext.String("provider")
+	providerGPUShape := cliContext.String("provider-gpu-shape")
+	providerInstanceType := cliContext.String("provider-instance-type")
+	xrayNeeded := cliContext.Bool("xray-needed")
+	totalCPU := cliContext.Int64("total-cpu")
+	extraInfo := cliContext.String("extra-info")
```

Comment:
Add TODOs to auto derive default values (e.g., GPU type), and also prompt the user to input if not set.